Rust DuckDB quickstart does now compile if using Arrow version 57

Hello, I am trying to use the ADBC driver for DuckDB in Rust. The current Rust DuckDB quickstart runs successfully, however, If I update `arrow` and `arrow-array` to the latest version (57.0.0), I get the following error:

```
error[E0277]: a value of type `Vec<arrow_array::RecordBatch>` cannot be built from an iterator over elements of type `arrow_array::record_batch::RecordBatch`
  --> duckdb/src/main.rs:44:54
   |
44 |     let batches: Vec<RecordBatch> = reader.collect::<Result<_, _>>().unwrap();
   |                                            -------   ^^^^^^^^^^^^ value of type `Vec<arrow_array::RecordBatch>` cannot be built from `std::iter::Iterator<Item=arrow_array::record_batch::RecordBatch>`
   |                                            |
   |                                            required by a bound introduced by this call
   |
   = help: the trait `FromIterator<arrow_array::record_batch::RecordBatch>` is not implemented for `Vec<arrow_array::RecordBatch>`
           but trait `FromIterator<arrow_array::RecordBatch>` is implemented for it
   = help: for that trait implementation, expected `arrow_array::RecordBatch`, found `arrow_array::record_batch::RecordBatch`
   = note: required for `Result<Vec<arrow_array::RecordBatch>, arrow_schema::error::ArrowError>` to implement `FromIterator<Result<arrow_array::record_batch::RecordBatch, arrow_schema::error::ArrowError>>`
```

I tried changing `use arrow_array::RecordBatch;` to `use arrow_array::record_batch::RecordBatch;`, but this gives me a `module 'record_batch' is private` error.

Admittedly, I am quite new to Rust so could be making some very basic mistake. Any help would be greatly appreciated!

--- a/rust/bigquery/Cargo.toml
+++ b/rust/bigquery/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/duckdb/Cargo.toml
+++ b/rust/duckdb/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/flightsql/Cargo.toml
+++ b/rust/flightsql/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/mssql/Cargo.toml
+++ b/rust/mssql/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/mysql/Cargo.toml
+++ b/rust/mysql/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/postgresql/Cargo.toml
+++ b/rust/postgresql/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/redshift/Cargo.toml
+++ b/rust/redshift/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/snowflake/Cargo.toml
+++ b/rust/snowflake/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

--- a/rust/sqlite/Cargo.toml
+++ b/rust/sqlite/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow-array = "56.2.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow-array = "57.0.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }

--- a/rust/trino/Cargo.toml
+++ b/rust/trino/Cargo.toml
@@ -18,7 +18,7 @@ version = "0.1.0"
 edition = "2024"
 
 [dependencies]
-adbc_core = "0.20.0"
-adbc_driver_manager = "0.20.0"
-arrow = { version = "56.2.0", features = ["prettyprint"] }
-arrow-array = "56.2.0"
+adbc_core = "0.21.0"
+adbc_driver_manager = "0.21.0"
+arrow = { version = "57.0.0", features = ["prettyprint"] }
+arrow-array = "57.0.0"

